### ISSUE TITLE
Error when switching from "listMonth" to "dayGridMonth" with "showNonCurrentDates" disabled

### ISSUE THREAD

**Issue by qbuache** [fullcalendar/fullcalendar#4677]
Hello,

An error occurs when switching from `listMonth` directly to `dayGridMonth` when `showNonCurrentDates` is set to `false`.

It does not occur when switching to an other view first.

#### How to reproduce 
Click on the `dayGridMonth` button when on `listMonth` view.
https://codepen.io/anon/pen/RmaZLe?editors=0010

CodePen and my website don't show the exact same stacktrace.

CodePen : 
``` 
main.min.js:8 Uncaught TypeError: Cannot read property 'valueOf' of undefined
    at e.toDate (VM2169 main.min.js:8)
    at n._renderCells (main.min.js:20)
    at n.i [as renderCells] (VM2169 main.min.js:6)
    at n.render (main.min.js:20)
    at n.e.receiveProps (VM2169 main.min.js:8)
    at t.render (main.min.js:20)
    at t.e.receiveProps (VM2169 main.min.js:8)
    at n.render (main.min.js:20)
    at n.e.receiveProps (VM2169 main.min.js:8)
    at n.renderView (VM2169 main.min.js:8)
```

My website :
```
main.min.js:6 Uncaught TypeError: Cannot read property 'getUTCFullYear' of undefined
    at ie (main.min.js:6)
    at e.toDate (main.min.js:8)
    at n._renderCells (main.min.js:20)
    at n.i [as renderCells] (main.min.js:6)
    at n.render (main.min.js:20)
    at n.e.receiveProps (main.min.js:8)
    at t.render (main.min.js:20)
    at t.e.receiveProps (main.min.js:8)
    at n.render (main.min.js:20)
    at n.e.receiveProps (main.min.js:8)
```
#### Info
Using version 4.1.0 without Scheduler
Does not happen in version 3

Have a nice day

**Comment by acerix** [comment]
Thanks for the detailed report, confirmed.

**Comment by sidouglas** [comment]
@qbuache thank you for this - I ran into this same problem today. I thought I was losing my mind.

**Comment by arshaw** [comment]
This was fixed in 4.3.0 though please start using 4.3.1
Releases: https://github.com/fullcalendar/fullcalendar/releases
